internal/oc/release.go

```diff
@@ -126,7 +127,16 @@ func (r *release) GetMustGatherImage(log logrus.FieldLogger, releaseImage string
 
 // GetCoreOSImage gets rhel-coreos image URL from the release image or releaseImageMirror, if provided.
 func (r *release) GetCoreOSImage(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error) {
-	return r.getImageByName(log, coreosImageName, releaseImage, releaseImageMirror, pullSecret)
+	imageName := coreosImageName
+	if isOKD(releaseImage) {
+		imageName = streamCoreOSImageName
+	}
+	return r.getImageByName(log, imageName, releaseImage, releaseImageMirror, pullSecret)
+}
+
+// Returns true if the image matches OKD release image pattern
+func isOKD(image string) bool {
+	return strings.Contains(image, "okd")
```

Comment:
this is checking the pullspec string right? so if a pullspec with digest is provided, this won't work?

Reply:
Yes, what you say is correct - this should be changed.
Currently we check for the image tagged `okd-rpms` within the release image (`oc adm release info --image-for=okd-rpms --insecure=false <release image>`) - however this seems that it's not working any longer from release 4.16 onward. This would mean OKD assisted installs for OCP > 4.15 are not working properly. Could you please confirm that this method is no longer valid?

Reply:
right...we don't build the okd-rpms image anymore. The openshift binaries are pulled from ART's mirror (for now).

Reply:
you could check for `stream-coreos` instead as that is specific to OKD.